app/src/pages/playground/types.ts

```diff
@@ -1,7 +1,18 @@
+import { spanPlaygroundPageLoaderQuery$data } from "./__generated__/spanPlaygroundPageLoaderQuery.graphql";
+
 export interface PlaygroundInstanceProps {
   /**
    * Multiple playground instances are supported.
    * The id is used to identify the instance.
    */
   playgroundInstanceId: number;
 }
+
+/**
+ * The type of a span that is fetched to pre-populate the playground.
+ * This span gets fetched when navigating from a span to the playground, used for span replay.
+ */
+export type PlaygroundSpan = Extract<
+  spanPlaygroundPageLoaderQuery$data["span"],
+  { __typename: "Span" }
+>;
```

Comment:
Since this is so tightly coupled with the loader, I wonder if we should just put it there - it's more honest.

Reply:
yeah i just didnt' love the idea of exporting types from component files because they get imported into utils / constants files which leads to circularness, could also just define it in the utils file since thats really the only place it's needed